src/common/src/error.rs

```diff
@@ -337,18 +276,6 @@ impl From<String> for RwError {
     }
 }
 
-impl Debug for RwError {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        write!(
-            f,
-            "{}\n{}",
-            self.inner,
-            // Use inner error's backtrace by default, otherwise use the generated one in `From`.
-            std::error::request_ref::<Backtrace>(&self.inner).unwrap_or(&*self.backtrace)
-        )
-    }
-}
```

Comment:
Is this part the user-facing change? i.e., "As a result, we might have to review all (important) usages of error logging and replace them with thiserror_ext::Report."

Can you elaborate a bit more about the changes?

Reply:
Strictly speaking it's not as the only user-facing part is the psql interface. However, it really impacts the developing/debugging experience as we print logs for errors. 😕

Reply:
By "user-facing" I don't mean end-user, but just wanted to know what's the behavior change.

Reply:
i.e., for users of the error type. Is Display and/or Debug and/or other things changed.

Reply:
Just want to confirm that so that we can better evaluate the radius of influence

Reply:
Wait, does `Debug` contain the source chain now? (and should it?) !

Reply:
IIRC `anyhow`'s `Debug` is still relatively pretty

Reply:
https://github.com/dtolnay/anyhow/blob/05e413219e97f101d8f39a90902e5c5d39f951fe/src/fmt.rs#L7-L65

Can we copy this implementation? 😄

Reply:
I'm not blocking this PR if `Debug` is not readable now. Either change to `report` or impl better `Debug` in the future is OK to me.

But I'd like to double-confirm these (even if I *feel* it's the case):
- `Debug` is *loseless*
- `Display` is not changed

Reply:
> IIRC `anyhow`'s `Debug` is still relatively pretty

This is true but I'm not sure whether we should adopt the same idea. `anyhow::Error` can be used throughout an application, so developers may get used to the debug formatting and use it everywhere.

However, not all errors are wrapped with new types in RisingWave, and developers may also log the external errors directly. That is to say, developers still have to choose different formatting based on the concrete type. In this sense, `as_report` can be a more unified solution.

Reply:
> * `Display` is not changed

Yes.

> * `Debug` is _loseless_

I guess it's lossy. Backtrace is lost and it'll use the derived `Debug` impl. However I suddenly realize that the refactor does not conflict with customized `Debug` implementation. Let's revert it now and redo it after #13750.

Reply:
My point is that: in the perfect world, `Debug` should never be used at all! But in fact developers are not knowledgeable or careful enough, so they may misuse `Debug`. That's why I think `Debug` should contain enough information - to make it more foolproof. Not that related with the difference between application and library.